src/App.tsx

```diff
@@ -8,37 +8,142 @@ import { PostsList } from './components/PostsList';
+
+    if (userId === selectedUserId) {
+      return;
+    }
+
+    setIsLoading('users');
+    setSelectedUserId(userId);
+
+    getUserPosts(userId)
+      .then((userPostsFromServer) => {
+        if (userPostsFromServer.length === 0) {
+          setIsError(Error.NOPOSTS);
+        }
+
+        setSelectedUserPosts(userPostsFromServer);
+      })
+      .catch(() => {
+        setIsError(Error.USERS);
+        setSelectedUserPosts([]);
+      })
+      .finally(() => setIsLoading(''));
+  };
+
+  const handleCommentsLoads = (postId: number) => {
+    setIsLoading('comments');
+    setIsError(Error.NONE);
+
+    getPostComments(postId)
+      .then(commentsFromServer => setPostComments(commentsFromServer))
+      .catch(() => setIsError(Error.COMMENTS))
+      .finally(() => setIsLoading(''));
+  };
+
+  const handlePostSelect = (postId: number) => {
+    if (postId !== selectedPostId) {
+      setFormIsActive(false);
+    }
+
+    setSelectedPostId(postId);
+    handleCommentsLoads(postId);
+  };
+
+  const handleAddComment = (newComment: Comment) => {
+    setPostComments([...postComments, newComment]);
+  };
+
+  const selectedPost
+    = selectedUserPosts.find(post => post.id === selectedPostId);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUserId={selectedUserId}
+                  onSelectUser={handleUserSelect}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!isError && selectedUserId === 0 && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading === 'users' && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isError === Error.USERS && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {isError}
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {
+                  !isLoading
+                  && isError === Error.NOPOSTS
+                  && selectedUserId > 0
+                  && selectedUserPosts.length === 0
```

Comment:
This can be replaced with `!selectedUserPosts.length` as  [0 represents false](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)